### PR TITLE
Fix game crash when alt tabbing

### LINKED ISSUE #24528: Crash when minimizing or alt-tabbing to the desktop
Game version: 0.C-29754-g34eea34 (console) (build 7627, latest experimental)

Operating system: Windows 8.1 64bit

Tiles or curses: curses (but tiles still has problems with screen resizing)

Mods active: any

### Expected behavior

No crash, minimizes/alt-tabs fine.

### Actual behavior

gn7mdnv

5cnysz7

### Steps to reproduce the behavior

Launch the game then minimize the game or alt-tab to the desktop (alt-tabbing to other windows is fine).

### Other notes

After the crash the terminal width and height returns to the defaults 80 and 24 regardless of what it was to before. (The game also crashes and behaves erratically when trying to resize the window)
The log doesn't produce anything useful.

### PR DESCRIPTION
**Pull request by 8street**
#### Summary
```SUMMARY: Bugfixes "Fix game crash when alt tabbing"```

#### Purpose of change
Fixes #24528
Now when alt-tabbing game not crash in fullscreen mode.

#### Describe the solution
After alt-tabbing the game window takes a very small wrong resolution which can lead to crash.  Now the game does not resize its screen when alt-tabbing at all.

#### Describe alternatives you've considered
After alt-tabbing it would be possible to return screen resolution that was before alt-tabbing. But it works unstable. Sometimes it leads to texture nullptr.

#### Testing
Tested in Win 10 64bit. Tiles with sounds.
Alt-tabbing in game and in main menu when fullscreen mode activated.
Alt-tabbing and resized in not fullscreen mode.

#### Additional context

### CODE REVIEW

**Comment by 1n17** [comment]
> Fixes #24528

Partially at best - this does nothing on wincurse port. And since curses and sdl experience the same problem in the exact same point something tells me at best you're sidestepping the issue.

**Comment by 8street** [comment]
Yes, this is a workaround. But it looks very transparent to the user. When a game is minimized, it becomes not in fullscreen mode. And this is not noticeable. When the game is restored, it again becomes in fullscreen mode.

**Comment by 8street** [comment]
I'm thinking of closing this pull request. The game is still not stable in full screen mode. Sometimes there are some bugs when loading the game. Especially if I switch the tiles in the options many times.
err

**Comment by ZhilkinSerg** [comment]
> 
> 
> I'm thinking of closing this pull request. The game is still not stable in full screen mode. Sometimes there are some bugs when loading the game. Especially if I switch the tiles in the options many times.
> err

Rebase your branch on top of most recent `master`, also remove `options.json` and test in a fresh world.

**Comment by 8street** [comment]
Thanks, it helped. I also found a more elegant solution to the problem. Now the game does not resize its screen at all. I hope nothing breaks on the android.

File: src/sdltiles.cpp [hunk 1]
```diff
@@ -2861,7 +2863,15 @@ static void CheckMessages()
                         break;
 #endif
                     case SDL_WINDOWEVENT_SHOWN:
+                        break;
                     case SDL_WINDOWEVENT_EXPOSED:
+                        break;
```
Comment (Night-Pryanik) [inline src/sdltiles.cpp]: These `break`s here are redundant. Both `SDL_WINDOWEVENT_SHOWN` and `SDL_WINDOWEVENT_EXPOSED` will auto-fallthough to `break` for `SDL_WINDOWEVENT_MINIMIZED`.

Comment (8street) [reply src/sdltiles.cpp]: Mentioned breaks deleted.

**Pull request merged**